src/main/java/org/candlepin/subscriptions/event/EventController.java

```diff
@@ -169,15 +163,22 @@ private Map<EventKey, Event> parseEventRecordsToEventsEntityMap(Set<String> even
                   "Ensuring orgId={} has been set up for syncing/reporting.", event.getOrgId());
               ensureOptIn(event.getOrgId());
             }
-            eventsMap.putIfAbsent(EventKey.fromEvent(event), event);
+
+            if (event.getAction() != null && event.getAction() == Event.Action.CLEANUP) {
```

Comment:
Nothing wrong but a suggestion that you may consider changing here:
Instead of !=null check you can use Objects.isnull(event.getAction()) for readability. 
Also for enum checking you can use equals. However, since Event.Action is an enum and there's only one instance of Event.Action.ADD (i.e., it's effectively a singleton) I don't mind what you have done here.

Reply:
Honestly, I read better obj != null rather than Objects.isNull(obj). But I have no objections if this is a team decision-wise.

Reply:
I am fine either ways as mentioned above its a suggestion.